### PR TITLE
Javascript support

### PR DESCRIPTION
**Pull request by StarveTheEgo**
Here is my humble Javascript port attempt (from Python). Seems to be working

There are a lot of 'await' keywords, i am not really sure, is it even cool here.

Also there is some custom-made 30-s timeout for reading socket data, looks weird for me (if you connect to the server, but server will not Start the game in 30seconds - an error will be thrown, meh)

Copied DOCKERFILE and install.sh from Javascript client from 2018's cup

### CODE REVIEW

File: clients/javascript/model/game.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        this.mines = mines;
+        this.lootBoxes = lootBoxes;
+    }
+
+    static async readFrom (stream) {
+        const currentTick = await stream.readInt();
+        const properties = await Properties.readFrom(stream);
+        const level = await Level.readFrom(stream);
+
+        let i, arraySize;
```
Comment (StarveTheEgo) [inline clients/javascript/model/game.js:26]: Probably here i should duplicate **i** and **arraySize** to make it code generator friendly

**Comment by StarveTheEgo** [comment]
Also probably in order to make it generator-friendly we should duplicate those 'require' calls before each class declaration, should we?

**Comment by kuviman** [comment]
@StarveTheEgo if this works, adopting to code generation shouldn't be hard

**Comment by StarveTheEgo** [comment]
Added manual weapon reloading changes too

**Comment by kuviman** [comment]
Thanks! Some things weren't working here, so it took a bit more time than expected.

JavaScript added now

**Pull request closed without merging**